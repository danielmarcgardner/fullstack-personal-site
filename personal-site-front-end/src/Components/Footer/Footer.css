#footer {
  width: 100%;
  height: 50px;
  background-color: #202938;
}

#copyright {
  display: inline-block;
  color: #679ffc;
  text-align: center;
  height: 50px;
  padding-top:15px;
  margin-left: 50px;
}

#footerlinks {
  float:right;
}
