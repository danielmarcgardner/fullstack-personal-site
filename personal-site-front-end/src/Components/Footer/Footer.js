import React from 'react';
import './Footer.css';

const Footer = props => (
  <div id="footer">
    <div id="copyright">
      &copy; Daniel Gardner 2017
    </div>
    <div id="footerlinks">
      <div className="navitem" onClick={() => props.history.push('/signup')}>
      Sign Up
    </div>
      <div className="navitem" onClick={() => props.history.push('/signin')}>
      Sign In
    </div>
    </div>
  </div>
);

export default Footer;
