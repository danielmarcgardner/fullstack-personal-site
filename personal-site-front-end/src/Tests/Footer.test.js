import React from 'react';
import { shallow } from 'enzyme';
import Footer from '../Components/Footer/Footer';

const component = shallow(<Footer />);

describe('Footer', () => {
  it('Renders', () => {
    expect(component).toMatchSnapshot();
  });
  it('Has a copyright', () => {
    expect(component.find('#copyright').length).toEqual(1);
  });
  it('Has footerlinks', () => {
    expect(component.find('#footerlinks').length).toEqual(1);
  });
  it('Has two navitems', () => {
    expect(component.find('.navitem').length).toEqual(2);
  });
});
