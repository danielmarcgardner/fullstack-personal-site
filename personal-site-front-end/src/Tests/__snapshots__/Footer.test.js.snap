// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Renders 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    id="footer"
>
    <div
        id="copyright"
    >
        © Daniel Gardner 2017
    </div>
    <div
        id="footerlinks"
    >
        <div
            className="navitem"
            onClick={[Function]}
        >
            Sign Up
        </div>
        <div
            className="navitem"
            onClick={[Function]}
        >
            Sign In
        </div>
    </div>
</div>,
  "nodes": Array [
    <div
      id="footer"
>
      <div
            id="copyright"
      >
            © Daniel Gardner 2017
      </div>
      <div
            id="footerlinks"
      >
            <div
                  className="navitem"
                  onClick={[Function]}
            >
                  Sign Up
            </div>
            <div
                  className="navitem"
                  onClick={[Function]}
            >
                  Sign In
            </div>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Footer />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          id="footer"
>
          <div
                    id="copyright"
          >
                    © Daniel Gardner 2017
          </div>
          <div
                    id="footerlinks"
          >
                    <div
                              className="navitem"
                              onClick={[Function]}
                    >
                              Sign Up
                    </div>
                    <div
                              className="navitem"
                              onClick={[Function]}
                    >
                              Sign In
                    </div>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          id="footer"
>
          <div
                    id="copyright"
          >
                    © Daniel Gardner 2017
          </div>
          <div
                    id="footerlinks"
          >
                    <div
                              className="navitem"
                              onClick={[Function]}
                    >
                              Sign Up
                    </div>
                    <div
                              className="navitem"
                              onClick={[Function]}
                    >
                              Sign In
                    </div>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Footer />,
}
`;
